distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/ServerProcessBuilder.java

```diff
@@ -150,20 +151,18 @@ ServerProcess start(ProcessStarter processStarter) throws UserException {
 
         Process jvmProcess = null;
         ErrorPumpThread errorPump;
+        PumpThread stdoutPump;
 
         boolean success = false;
         try {
             jvmProcess = createProcess(getCommand(), getJvmArgs(), jvmOptions, getEnvironment(), processStarter);
-            errorPump = new ErrorPumpThread(terminal.getErrorWriter(), jvmProcess.getErrorStream());
+            errorPump = PumpThread.stderrPump(terminal, jvmProcess);
+            stdoutPump = PumpThread.stdoutPump(terminal, jvmProcess);
             errorPump.start();
+            stdoutPump.start();
             sendArgs(serverArgs, jvmProcess.getOutputStream());
 
-            String errorMsg = errorPump.waitUntilReady();
```

Comment:
Removed, seems to be dead code

Reply:
I don't think this is dead code. When we receive the error signal through stderr, the process may not have exited yet. So we wait for it to finish in order to get the exit code.

Reply:
Sorry, I highlighted the wrong lines... I was referring to `errorMsg` and the following if clause. `errorPump.waitUntilReady();` has to be kept of course, but it can simply return void (instead of `null`)

Reply:
Currently you have waitUntilReady rethrowing exceptions. That would not wait for the forked process to exit, nor convert that exception into a UserException as the existing code does. Am I missing something?

Reply:
I've updated the code based on your changes, thanks for the fix 👍